### PR TITLE
enhancement: support panic receipts

### LINKED ISSUE #668: Add support for `Panic` receipt
We don't yet support indexing for [`Panic` receipts](https://github.com/FuelLabs/fuel-specs/blob/master/src/protocol/abi/receipts.md#panic-receipt); we should add it to the requisite places in our codegen.

### PR DESCRIPTION
**Pull request by rvmelkonian**
Closes #668 

This PR allows an indexer to index and capture Panic Receipts generated from chain. 

#Changelog 
• Updates codegen with new Receipt arm. 
• Updates documentation. 
• Creates new test.

## Testing
1. The new test `test_can_trigger_and_index_panic_function_postgres` should pass, along with other integration tests. 
2. Check the new documentation `cd docs && mdbook serve`.

## Notes
I had to use `fuel_tx::PanicReason` for the `reason` field in `abi::Panic` as `reason.reason()` returns `&PanicReason`, using `u64` throws an error.

### CODE REVIEW

File: packages/fuel-indexer-macros/src/indexer.rs [hunk 1]
```diff
@@ -509,6 +509,12 @@ fn process_fn_items(
                                 let data = bincode::serialize(&abi::TransferOut{ contract_id: id, to, asset_id, amount, pc, is }).expect("Bad encoding,");
                                 decoder.decode_type(ty_id, data);
                             }
+                            Receipt::Panic { id, reason, .. } => {
+                                #contract
+                                let ty_id = abi::Panic::type_id();
+                                let data = bincode::serialize(&abi::Panic{ contract_id: id, reason: *reason.reason() }).expect("Bad encoding,");
```
Comment (deekerno) [inline packages/fuel-indexer-macros/src/indexer.rs:515]: See comment about using a regular number type rather than `PanicReason`.

**Review by deekerno — CHANGES_REQUESTED** [review]
Looking good! Just a few comments, but the functionality is all there. 👍🏽

File: packages/fuel-indexer-types/src/abi.rs [hunk 2]
```diff
@@ -156,6 +157,18 @@ impl NativeFuelType for Call {
     }
 }
 
+#[derive(Deserialize, Serialize, Debug, Clone)]
+pub struct Panic {
+    pub contract_id: ContractId,
+    pub reason: PanicReason,
+}
```
[the remark this answers is not in the document]
Comment (rvmelkonian) [reply packages/fuel-indexer-types/src/abi.rs:164]: yep, good idea.

File: docs/src/reference-guide/indexing/receipts/panic.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+use fuel_types::ContractId;
+use fuel_tx::PanicReason;
+pub struct Panic {
+    pub contract_id: ContractId, 
+    pub reason: PanicReason, 
+}
+```
+
+- A `Panic` receipt is produced when a Sway smart contract call fails for a reason that doesn't produce a revert. 
+- The reason is type PanicReason, which is an enum of `u8` variants, you can see the reason values
```
[the remark this answers is not in the document]
Comment (rvmelkonian) [reply docs/src/reference-guide/indexing/receipts/panic.md:13]: Yes. Thanks you.

**Review by deekerno — APPROVED** [review]
:shipit:

**Pull request merged**